omnigibson/simulator.py

```diff
@@ -474,8 +475,9 @@ def _transition_rule_step(self):
         for added_obj_attr in added_obj_attrs:
             new_obj = added_obj_attr.obj
             self.import_object(added_obj_attr.obj)
-            pos, orn = added_obj_attr.pos, added_obj_attr.orn
-            new_obj.set_position_orientation(position=pos, orientation=orn)
+            new_obj.set_position_orientation(position=added_obj_attr.pos, orientation=added_obj_attr.orn)
+            if isinstance(new_obj, DatasetObject):
+                new_obj.set_bbox_center_position_orientation(position=added_obj_attr.bb_pos, orientation=added_obj_attr.bb_orn)
```

Comment:
This seems a bit too hacky. Is all dataset object using the set_bbx_center_position_orientation mechanism?

Reply:
To a bit of sanity check here to safeguard added_obj_attr.attr exists.

Reply:
it's a either/or situation, not an overwritten situation.

Reply:
done!